test/whisperkit_test.py

```diff
@@ -0,0 +1,190 @@
+
+class TestWhisperKitAndroid(AndroidTestsMixin):
+    OUTPUT_FOLDER = './json/'   
+    TEST_DATASET = "librispeech-10mins" 
+
+    @classmethod
+    def setUpClass(self):
+        super().setUpClass()
+
+    def __init__(self, methodName='runTest', args=None):
+        super().__init__(methodName)
+        self.args = args
+        self.test_bin = "whisperax_cli"
+        self.root_path = "/sdcard/argmax/tflite"
+        self.devices = []
+        self.tokenizer = get_tokenizer(multilingual=False, language="en")
+
+        if self.args.model_path.find("openai_whisper-tiny") != -1:
+            self.model_size = "tiny"
+        elif self.args.model_path.find("openai_whisper-base") != -1:
+            self.model_size = "base"
+        elif self.args.model_path.find("openai_whisper-small") != -1:
+            self.model_size = "small"
+
+        output = subprocess.run(["adb", "devices"], stdout=subprocess.PIPE)
+        if output.stdout is None:
+            return None
+        devices = output.stdout.decode('utf-8')
+        words = devices.split()
+
+        attached_found = False
+        for index, word in enumerate(words):
+            if attached_found is False:
+                if word.find('attached') != -1:
+                    attached_found = True
+                else:
+                    continue
+
+            if len(words) > index + 1:
+                if words[index + 1].find("device") != -1:
+                    self.devices.append(word)
+
+        splitted = os.path.split(os.path.abspath(self.args.input))
+        path_parts = Path(self.args.input).parts
+        if os.path.isfile(self.args.input):
+            self.path = splitted[0]
+            self.files = [splitted[1]]
+            self.data_set = path_parts[-2]
+        
+        elif os.path.isdir(self.args.input):
+            self.path = self.args.input
+            self.data_set = path_parts[-1]
+            self.files = []
+
+        metadata_path = os.path.join(self.path, "metadata.json")
+        metadata_file = open(metadata_path)
+        self.metadata = json.load(metadata_file)
+        metadata_text = json.dumps(self.metadata)
+        metadata_file.close()
+
+        if os.path.isdir(self.args.input):
+            for file in os.listdir(self.args.input):
+                full_file_name = os.path.join(self.args.input, file)
+                # checking if it is a file
+                if os.path.isfile(full_file_name) is False:
+                    continue
+                
+                name_only, extension = os.path.splitext(file)
+                if extension not in self.audio_file_ext:
+                    continue
+
+                if metadata_text.find(name_only) == -1:
+                    continue
+
+                self.files.append(file)
+
+    def test_audio_transcription(self):
+        if os.path.exists(TestWhisperKitAndroid.OUTPUT_FOLDER) is False:
+            os.mkdir(TestWhisperKitAndroid.OUTPUT_FOLDER)
```

Comment:
I believe you can use os.mkdirs() instead of os.mkdir and skip the existence check